docs/proposals/32-version-classification-lifecycles.md

```diff
@@ -0,0 +1,282 @@
+
+With this change we also introduce a resource status for the `CloudProfile` to improve the issue that the actual classification stage is not immediately obvious. The status makes it more readable for API consumers and they do not need to calculate the actual classification stage on their own.
+
+```yaml
+# assume that the current date is 2024-12-03
+apiVersion: core.gardener.cloud/v1beta1
+kind: CloudProfile
+metadata:
+  name: local
+spec:
+  kubernetes:
+    versions:
+      - version: 1.30.6
+        lifecycle:
+          - classification: preview # starts in preview because no start time is defined
+          - classification: supported
+            startTime: "2024-12-01T00:00:00Z"
+          - classification: deprecated
+            startTime: "2025-03-01T00:00:00Z"
+          - classification: expired
+            startTime: "2025-04-01T00:00:00Z"
+status:
+  kubernetes:
+    versions:
+      - version: 1.30.6
+        classificationState: supported
+```
+
+In addition to the existing classification stages, we add one more stage with the name `unavailable` to the API. An `unavailable` version is planned to become available in the future. It is not possible to reference this version in this stage and can be used by administrators to schedule a new version release.
+
+The `expired` classification, which existed only implicitly in the API, now becomes a dedicated value.
+
+So, the resulting list of classification stages will be:
+
+- `unavailable`
+- `preview`
+- `supported`
+- `deprecated`
+- `expired`
+
+There are rules for the new lifecycle, some of them need to be ensured through validations:
+
+- Classification stages in a lifecycle must only appear ordered from `unavailable` to `expired` as described in the list above.
+- It is not required that every classification stage is present in the lifecycle.
+- Start times are always monotonically increasing.
+- The first start date is optional and interpreted as zero time, meaning it has already started.
+- If no lifecycle is given, it defaults to a lifecycle definition with one `supported` stage.
+- If all start times are in the future, the resulting classification is `unavailable`.
+
+There is already a controller in place for reconciling the `CloudProfile` (by now it's primarily handling finalizers only), which is going to be extended by reconciling the version classification statuses. If there are remaining stages inside `lifecycle` the next reconcile needs to be scheduled at its `startTime`.
+
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: CloudProfile
+metadata:
+  name: local
+spec:
+  kubernetes:
+    versions:
+      # if an administrator deploys just the version without any lifecycle,
+      # the reconciler will evaluate the classification status to supported
+      - version: 1.27.0
+
+      # when introducing a new version it may not contain any deprecation or expiration date yet
```

Comment:
Hmm...? I thought the idea is to allow adding new entries/versions with fully spec'ed out lifecycle stages, so why would it not be allowed to "contain any deprecation or expiration date"?

Or is this an English issue and should it actually be:
```suggestion
      # when introducing a new version it doesn't have to contain a deprecation or expiration date
```

Reply:
Stumbled upon the same but inferred that the meaning was rather that depr. or exp. is optional. So your reworded suggestion makes this definitely more clear (if it is the intended behaviour ☺️).

Reply:
Yeah, probably an issue with our English skills. 😅
It's not necessarily required to specify a deprecation or expiration date upfront.